docs/rules/max-lines.md

```diff
@@ -0,0 +1,117 @@
+# enforce a maximum file length (max-lines)
+
+Some people consider large files a code smell. Large files tend to do a lot of things and can make following what's going on hard.
+
+## Rule Details
+
+This rule enforces a maximum number of lines per file, in order to aid in maintainability and reduce complexity.
+
+
+## Options
+
+This rule has a number or object option:
+
+* `"max"` (default `100`) enforces a maximum number of lines in a file
```

Comment:
That's an interesting question for larger discussion-- does JSCS compatibility mean match JSCS as closely as possible to make integration require less effort, or is ESLint allowed to have a different opinion on defaults and "JSCS compatibility" merely means there is a way to run your old JSCS settings in ESLint (even with different/non-default configuration)?